jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -199,16 +201,84 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
     }
 
     /**
-     * Create an error diagnostic if a ServerEndpoint annotation does not follow the rules.
+     * Creates an error diagnostic if there exists more than one method annotated
+     * with @OnMessage for a given message format.
+     * 
+     * @param type
+     * @param diagnostics
+     * @param unit
+     * @throws JavaModel
+     */
+    private void onMessageWSMessageFormats(IType type, List<Diagnostic> diagnostics, ICompilationUnit unit)
+            throws JavaModelException {
+        IMethod[] typeMethods = type.getMethods();
+        IAnnotation onMessageTextUsed = null;
+        IAnnotation onMessageBinaryUsed = null;
+        IAnnotation onMessagePongUsed = null;
+        for (IMethod method : typeMethods) {
+            IAnnotation[] allAnnotations = method.getAnnotations();
+            for (IAnnotation annotation : allAnnotations) {
+                if (annotation.getElementName().equals(WebSocketConstants.ON_MESSAGE)) {
+                    ILocalVariable[] allParams = method.getParameters();
+                    for (ILocalVariable param : allParams) {
+                        if (!isParamPath(param)) {
+                            String signature = param.getTypeSignature();
+                            String formatSignature = signature.replace("/", ".");
+                            String resolvedTypeName = JavaModelUtil.getResolvedTypeName(formatSignature, type);
+                            if (resolvedTypeName != null
```

Comment:
Is there a specific reason why when `resolvedTypeName ` is null, the analysis is ignored? I think we should obtain the parameter type in another way when the `resolvedTypeName` is null?

Reply:
That was added due to the null pointer exceptions that only happened during the unit tests. See #265

Reply:
I understand that but we should stay obtain the parameter type in another way. I used Signature to get the raw parameter type, maybe you can use something similar to that

Reply:
In what PR was that?

Reply:
See [line 126](https://github.com/eclipse/lsp4jakarta/blob/b49113e3ddd665404441f223d79f198dfdd482bd/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java#L126). PR was [#260](https://github.com/eclipse/lsp4jakarta/pull/260)